Azure/azure-cli
# [Bot Service] BREAKING CHANGE: `az bot create` change: Remove `--kind`, `--password`, `--lang` arguments. Add `--app-type`, `--tenant-id`, `--msi-resource-id` arguments

**Related command**
az bot create

**Description**
The function of `az bot create` command is not corresponding with the azure devportal UI. So we plan to change the command parameters the same as devportal. This pr is to change the parameters same as 'Use existing app registration' type on devportal.

**Testing Guide**

Newly added test cases:

- test_botservice_registration_bot_create_singletenant_type
- test_botservice_registration_bot_create_multitenant_type
- test_botservice_registration_bot_create_userassignedmsi_type

Manually validate existing functionality

- run az bot create -g {} -n {} --appid {} --app-type SingleTenant --tenant-id {} -l {}
- run az bot create -g {} -n {} --appid {} --app-type MultiTenant -l {}
- run az bot create -g {} -n {} --appid {} --app-type UserAssignedMSI --tenant-id {} --msi-resource-id {} -l {}

---

This checklist is used to make sure that common guidelines for a pull request are followed.
